space/space_test.go

```diff
@@ -263,3 +265,59 @@ func (s *repoBBTest) TestLoadByOwnerAndName() {
 		require.Nil(t, sp)
 	})
 }
+
+func (s *SpaceRepositoryTestSuite) TestLoadMany() {
+	s.T().Run("ok", func(t *testing.T) {
+		// given
+		fxt := tf.NewTestFixture(t, s.DB, tf.Spaces(2))
+		ids := make([]uuid.UUID, 0)
+		for _, s := range fxt.Spaces {
+			ids = append(ids, s.ID)
+		}
+		// when listing
+		result, err := s.repo.LoadMany(s.Ctx, ids)
+		// then
+		require.Nil(t, err)
+		assert.Condition(t, containsAllSpaces(t, fxt.Spaces, result...))
+	})
+	s.T().Run("ok with duplicates", func(t *testing.T) {
+		// given
+		fxt := tf.NewTestFixture(t, s.DB, tf.Spaces(10))
+		ids := make([]uuid.UUID, 0)
+		// add the IDs twice
+		for _, s := range fxt.Spaces {
+			ids = append(ids, s.ID)
+			ids = append(ids, s.ID)
+		}
+		// when listing
+		result, err := s.repo.LoadMany(s.Ctx, ids)
+		// then make sure the result does not contain duplicates
+		require.Nil(t, err)
+		assert.Condition(t, containsAllSpaces(t, fxt.Spaces, result...))
```

Comment:
Can't this be simplified by doing:
```go
fxt := tf.NewTestFixture(t, s.DB, tf.Spaces(10))
// add ids twice
ids := []uuid.UUID{
  fxt.Spaces[0].ID, fxt.Spaces[0].ID,
  fxt.Spaces[1].ID, fxt.Spaces[1].ID,
  fxt.Spaces[2].ID, fxt.Spaces[2].ID,
}
// when listing
result, err := s.repo.LoadMany(s.Ctx, ids)
// then make sure the result does not contain duplicates
require.Nil(t, err)
expectedSpaces := map[uuid.UUID]struct{}{
   fxt.Spaces[0].ID: {},
   fxt.Spaces[1].ID: {},
   fxt.Spaces[2].ID: {},
}
for _ , s := range result {
  _, ok := expectedSpaces[s.ID]
  require.True(t, ok, "spaced %s not found in result set %'+v", s, result)
  // removal from the map ensures that the result contains spaces only once.
  delete(expectedSpaces, s.ID)
}
require.Empty(t, expectedSpaces)
```

I like this approach better because I don't have to learn how to use your `containsAllSpaces` function and the logic is applicable everywhere.

Reply:
well, I've seen this logic in other tests, but I can return the argument and say that I need to learn/understand why you have a list and remove items, etc. Besides, the `containsAllSpaces` returns an `assert.Comparison` which can be used in `assert.Condition()`. That and the function name make the statement quite self-expressive, IMHO.

Reply:
The `assert.Comparison` is just a function that returns a `bool`. So `assert.Condition()` will just report that something failed with no further detail. With my *map-element-removal-approach* (tm) you can also see what elements have not been included in the list because they still exist within the map and are printed.

Also, I find your logic in `containsAllSpaces` with nested for loops much more complicated.

IMHO unrolling a loop also gives more clarity in the test setup. That said, It is much easier to see what your intention is when you do

```go
ids := []uuid.UUID{
  fxt.Spaces[0].ID, fxt.Spaces[0].ID,
  fxt.Spaces[1].ID, fxt.Spaces[1].ID,
  fxt.Spaces[2].ID, fxt.Spaces[2].ID,
}
```
rather than
```go
ids := make([]uuid.UUID, 0)
 for _, s := range fxt.Spaces {
 	ids = append(ids, s.ID)
 	ids = append(ids, s.ID)
 }
```
If you like your approach better, please allocate enough memory before appending to your array.